File: dotnet/src/InternalUtilities/process/ProcessEvent.cs
// Copyright (c) Microsoft. All rights reserved.
using System.Runtime.Serialization;

namespace Microsoft.SemanticKernel.Process.Runtime;

/// <summary>
/// A wrapper around <see cref="KernelProcessEvent"/> that helps to manage the namespace of the event.
/// </summary>
/// <param name="Namespace">The namespace of the event.</param>
/// <param name="InnerEvent">The instance of <see cref="KernelProcessEvent"/> that this <see cref="ProcessEvent"/> came from.</param>
[DataContract]
public record ProcessEvent(
    [property: DataMember] string? Namespace,
    [property: DataMember] KernelProcessEvent InnerEvent)
{
    /// <summary>
    /// The Id of the event.
    /// </summary>
    internal string Id => $"{this.Namespace}.{this.InnerEvent.Id}";

    /// <summary>
    /// The data of the event.
    /// </summary>
    internal object? Data => this.InnerEvent.Data;

    /// <summary>
    /// The visibility of the event.
    /// </summary>
    internal KernelProcessEventVisibility Visibility => this.InnerEvent.Visibility;

    /// <summary>
    /// Creates a new <see cref="ProcessEvent"/> from a <see cref="KernelProcessEvent"/>.
    /// </summary>
    /// <param name="kernelProcessEvent">The <see cref="KernelProcessEvent"/></param>
    /// <param name="Namespace">The namespace of the event.</param>
    /// <returns>An instance of <see cref="ProcessEvent"/></returns>
    internal static ProcessEvent FromKernelProcessEvent(KernelProcessEvent kernelProcessEvent, string Namespace) => new(Namespace, kernelProcessEvent);
}

File: python/semantic_kernel/processes/process_event.py
# Copyright (c) Microsoft. All rights reserved.

from typing import Any

from semantic_kernel.kernel_pydantic import KernelBaseModel
from semantic_kernel.processes.kernel_process.kernel_process_event import (
    KernelProcessEvent,
    KernelProcessEventVisibility,
)
from semantic_kernel.utils.feature_stage_decorator import experimental


@experimental
class ProcessEvent(KernelBaseModel):
    """A wrapper around KernelProcessEvent that helps to manage the namespace of the event."""

    namespace: str | None = None
    inner_event: KernelProcessEvent

    @property
    def id(self) -> str:
        """The Id of the event."""
        return f"{self.namespace}.{self.inner_event.id}"

    @property
    def data(self) -> Any | None:
        """The data of the event."""
        return self.inner_event.data

    @property
    def visibility(self) -> "KernelProcessEventVisibility":
        """The visibility of the event."""
        return self.inner_event.visibility
